trunner/runners/ARMV7M7Runner.py

```diff
@@ -0,0 +1,161 @@
+
+#
+# Phoenix-RTOS test runner
+#
+# ARMV7M7 runner
+#
+# Copyright 2021 Phoenix SYstems
+# Authors: Jakub Sarzyński, Damian Loewnau
+#
+
+import time
+import sys
+import signal
+
+from .common import Psu, Phoenixd, PhoenixdError, PloError, PloTalker, Runner, DeviceRunner
+from .common import GPIO, logging, phd_error_msg, rootfs
+from trunner.tools.color import Color
+from pexpect.exceptions import TIMEOUT, EOF
+from abc import ABC, abstractmethod
+
+
+def handler(signum, frame):
+    logging.error("\nThe boot mode's change wasn't validated by pressing enter!\n")
+    sys.exit(1)
+
+
+class ARMV7M7Runner(DeviceRunner, ABC):
+    """This class provides interface to run tests on targets with armv7m7 architecture"""
+
+    status_color = {'BUSY': 'blue', 'SUCCESS': 'green', 'FAIL': 'red'}
+
+    def __init__(self, port, is_rpi_host):
+        super().__init__(port)
+        # default values, redefined by specified target runners
+        self.is_rpi_host = is_rpi_host
+        self.is_cut_power_used = False
+        self.flash_memory = 0
+
+        if self.is_rpi_host:
+            self.reset_gpio = GPIO(17)
+            self.reset_gpio.high()
+            self.power_gpio = GPIO(2)
+            self.power_gpio.high()
+            self.boot_gpio = GPIO(4)
+            self.leds = {'red': GPIO(13), 'green': GPIO(18), 'blue': GPIO(12)}
+
+            self.set_status(Runner.BUSY)
```

Comment:
this should be probably done in constructor of `Runner` class regardless or `is_rpi_host` as otherwise status is undefined

Reply:
Right, the set_status isn't needed here, when the status is set to `BUSY` in the Runner init

Reply:
still, You need to initialize status with a known value in constructor, as it should always have one of the defined values. 

In child class You can always call set_status second time after initializing GPIOS if needed (with some comment), but You may keep also changing state to BUSY in `test_runner.py`.